src/libraries/Common/src/Interop/Unix/System.Security.Cryptography.Native/README.md

```diff
@@ -0,0 +1,75 @@
+# System.Security.Cryptography.Native
+
+This folder contains C# bindings for native shim (libSystem.Security.Cryptography.Native.Openssl.so), shimming functionality provided by the OpenSSL library.
+
+## Memory allocation hooks
+
+One extra feature exposed by the native shim is tracking of memory used by
+OpenSSL by hooking the memory allocation routines via
+`CRYPTO_set_mem_functions`. This functionality is implemented in [native
+code](../../../../../../native/libs/System.Security.Cryptography.Native/memory_debug.c)
+because these memory callback can run in contexts where managed .NET code may
+not run (such as thread destructors).
+
+The functionality is enabled by setting
+`DOTNET_SYSTEM_NET_SECURITY_OPENSSL_MEMORY_DEBUG` to 1. This environment
+variable must be set before launching the program (calling
+`Environment.SetEnvironmentVariable` at the start of the program is not
+sufficient). The diagnostic API is not officially exposed and needs to be
+accessed via private reflection on the `Interop.Crypto` type located in the
+`System.Net.Security` assembly. On this type, you can use following static
```

Comment:
While System.Net.Security might expose it, System.Security.Cryptography feels more canonical to me (note the name of this library)